internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,34 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dockerHost = os.Getenv("DOCKER_HOST")
+		}
+		var bind string
+		if dockerHost == "" {
+			bind = "/var/run/docker.sock:/var/run/docker.sock"
+			if provider.os == "windows" {
+				bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+			}
+		} else {
+			switch {
+			case strings.HasPrefix(dockerHost, "tcp://"):
```

Comment:
This seems to say that `fd` can be handled the same way as `tcp`: https://stackoverflow.com/a/43408869

Reply:
@dfreilich I read http://0pointer.de/blog/projects/socket-activation.html just very briefly, it seems that `fd://` is special value for docker demon (when running as systemd service), not for docker client. For some reason client accepts it and treats `fd://` as `tcp://`.

Reply:
I changed behaviour. For any protocol that is not `unix://` nor `npipe://` the dockerHost flag is simply passed to build container via envvar. This should allow `fd://`.